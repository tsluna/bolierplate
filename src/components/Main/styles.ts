import styled from 'styled-components';

export const Wrapper = styled.main`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  background-color: #06092b;
  color: white;
  padding: 3rem;
  flex-direction: column;
`;

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
  color: white;
`;

export const Title = styled.h1`
  font-size: 2.5rem;
`;

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 3rem;
`;

export const Illustration = styled.img`
  width: min(30rem, 100%);
`;
