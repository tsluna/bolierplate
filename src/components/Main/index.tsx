import * as S from './styles';
const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de uma átomo e React Avançado escrito ao lado"
    />
    <S.Title>React Avançado</S.Title>
    <S.Description>
      TypeScript, ReactJs, NextJs e Styled Components
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Um desenvolvedor de frente para uma tela dom código"
    />
  </S.Wrapper>
);

export default Main;
