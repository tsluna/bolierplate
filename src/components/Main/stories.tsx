import { Meta, Story } from '@storybook/react/types-6-0';
import Main from '.';
export default {
  component: Main,
  title: 'Main',
  args: {
    title: 'Luna title',
    description: 'Minha descricao padrao'
  }
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;
export const Default: Story = (args) => <Main {...args} />;
