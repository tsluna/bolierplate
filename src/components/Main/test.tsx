import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  const { container } = render(<Main />);
  it('Should render the heading', () => {
    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
