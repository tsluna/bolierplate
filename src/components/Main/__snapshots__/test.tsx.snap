// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render the heading 1`] = `
<main
  class="styles__Wrapper-n3tl0o-0 gACAYW"
>
  <img
    alt="Imagem de uma átomo e React Avançado escrito ao lado"
    class="styles__Logo-n3tl0o-1 kyidDX"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-n3tl0o-2 hxmMzC"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-n3tl0o-3 dZPoVN"
  >
    TypeScript, ReactJs, NextJs e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela dom código"
    class="styles__Illustration-n3tl0o-4 bTktIU"
    src="/img/hero-illustration.svg"
  />
</main>
`;
